PurchasesTests/Purchasing/PurchasesTests.swift

```diff
@@ -1537,12 +1537,36 @@ class PurchasesTests: XCTestCase {
         expect(info).toEventuallyNot(beNil())
     }
 
-    func testWhenNoReceiptDataReceiptIsRefreshed() {
+    func testWhenNoReceiptReceiptIsRefreshed() {
         setupPurchases()
         self.receiptFetcher.shouldReturnReceipt = false
-        self.purchases?.restoreTransactions()
+        
+        makeAPurchase()
+        
+        expect(self.requestFetcher.refreshReceiptCalled).to(beTrue())
+    }
+
+    func testWhenNoReceiptDataReceiptIsRefreshed() {
+        setupPurchases()
+        self.receiptFetcher.shouldReturnReceipt = true
+        self.receiptFetcher.shouldReturnZeroBytesReceipt = true
+        
+        makeAPurchase()
+        
         expect(self.requestFetcher.refreshReceiptCalled).to(beTrue())
     }
+    
+    private func makeAPurchase() {
+        let product = MockSKProduct(mockProductIdentifier: "com.product.id1")
+        
+        // Second one issues an error
+        self.purchases?.purchaseProduct(product) { (tx, info, error, userCancelled) in }
```

Comment:
what does the comment mean? 
also, we should make sure that this doesn't no-op if purchases is nil, right? 
lastly, no need for explicit self, or even to name the variables in the block, so this is equivalent to
```swift
guard let purchases = purchases else { fatalError() }
purchases.purchaseProduct(product) { }
```

Reply:
however, if the transaction should be made after the product is purchased, then maybe the best way to do this would be to move the creation of the transaction and the storeKitWrapperDelegate call to inside the completion of purchaseProduct.

Reply:
The thing is that the `purchaseProduct` call will add the completion block to the array of completion blocks and we need to call `self.storeKitWrapper.delegate?.storeKitWrapper(self.storeKitWrapper, updatedTransaction: transaction)` in order to trigger the completion block of `purchaseProduct`.

In this case we don't care about the completion block of `purchaseProduct`, but we need to call it because it adds the payment to the "queue".

The weird comment is a copy paste issue